aws/request/retryer.go

```diff
@@ -35,10 +35,35 @@ type Retryer interface {
 }
 
 // WithRetryer sets a Retryer value to the given Config returning the Config
-// value for chaining.
+// value for chaining. The value must not be nil.
 func WithRetryer(cfg *aws.Config, retryer Retryer) *aws.Config {
```

Comment:
Is it appropriate to log a warning in such cases, where a value is set to nil and doing so might work but isn't recommended?